File: js/src/record_v2/getRecordV2.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getRecordV2 = void 0;
const sns_records_1 = require("@bonfida/sns-records");
const getRecordV2Key_1 = require("./getRecordV2Key");
const deserializeRecordV2Content_1 = require("./deserializeRecordV2Content");
/**
 * This function can be used to retrieve a specified record V2 for the given domain name
 * @param connection The Solana RPC connection object
 * @param domain The .sol domain name
 * @param record The record to search for
 * @returns
 */
function getRecordV2(connection_1, domain_1, record_1) {
    return __awaiter(this, arguments, void 0, function* (connection, domain, record, options = {}) {
        const pubkey = (0, getRecordV2Key_1.getRecordV2Key)(domain, record);
        const retrievedRecord = yield sns_records_1.Record.retrieve(connection, pubkey);
        if (options.deserialize) {
            return {
                retrievedRecord,
                deserializedContent: (0, deserializeRecordV2Content_1.deserializeRecordV2Content)(retrievedRecord.getContent(), record),
            };
        }
        return { retrievedRecord };
    });
}
exports.getRecordV2 = getRecordV2;

File: python/src/record_v2/get_record_v2.py
from solana.rpc.async_api import AsyncClient
from typing_extensions import TypedDict, NotRequired

from custom_types import Record
from record_v2.deserialize_record_content_v2 import deserialize_record_content_v2
from record_v2.get_record_key_v2 import get_record_key_v2
from sns_records.state import Record as SnsRecord


class GetRecordV2Options(TypedDict):
    deserialize: NotRequired[bool]


class RecordResult(TypedDict):
    retrieved_record: SnsRecord
    record: Record
    deserialized_content: NotRequired[str]


class SingleRecordResult(TypedDict):
    retrieved_record: SnsRecord
    deserialized_content: NotRequired[str]


async def get_record_v2(
    connection: AsyncClient,
    domain: str,
    record: Record,
    options: GetRecordV2Options = None,
) -> SingleRecordResult:
    """
    Retrieve a record from the Solana network.
    """
    pubkey = get_record_key_v2(domain, record)
    retrieved_record = await SnsRecord.retrieve(connection, pubkey)

    if options.get("deserialize", False):
        return SingleRecordResult(
            retrieved_record=retrieved_record,
            deserialized_content=deserialize_record_content_v2(
                retrieved_record.get_content(),
                record,
            ),
        )
    return SingleRecordResult(retrieved_record=retrieved_record)
